src/cargo/ops/cargo_update.rs

```diff
@@ -807,6 +749,60 @@ fn status_locking(ws: &Workspace<'_>, num_pkgs: usize) -> CargoResult<()> {
     Ok(())
 }
 
+fn required_rust_version(ws: &Workspace<'_>) -> Option<PartialVersion> {
+    if !ws.resolve_honors_rust_version() {
+        return None;
+    }
+
+    if let Some(ver) = ws.rust_version() {
+        Some(ver.clone().into_partial())
+    } else {
+        let rustc = ws.gctx().load_global_rustc(Some(ws)).ok()?;
```

Comment:
May not be worth a fix. I didn't benchmark it.

This will load and parse `.rustc_info.json` in a loop. May have some potential performance impact.

Reply:
Yeah. I had thought of that and had considered passing it around but wasn't sure if it was worth it.  We already are writing to the screen which is pretty slow.